src/main/java/com/scg/stop/proposal/domain/Proposal.java

```diff
@@ -0,0 +1,144 @@
+
+
+    private Proposal(User user, String title, String projectTypes, String email, String website, String content,
+                     Boolean isVisible, Boolean isAnonymous) {
+        this.title = title;
+        this.projectTypes= projectTypes;
+        this.email = email;
+        this.webSite = website;
+        this.content = content;
+        this.isAnonymous = isAnonymous;
+        this.isVisible = isVisible;
+        this.user = user;
+    }
+
+    public static Proposal createProposal(User user, String title, String projectTypes, String email, String website,
+                                           String content, Boolean isVisible, Boolean isAnonymous) {
+        return new Proposal(user, title, projectTypes, email, website, content, isVisible, isAnonymous);
+    }
+
+    public void update(String title, String projectTypes, String email, String website, String content,
+                       Boolean isVisible, Boolean isAnonymous) {
+        this.title = title;
+        this.projectTypes = projectTypes;
+        this.email = email;
+        this.webSite = website;
+        this.content = content;
+        this.isAnonymous = isAnonymous;
+        this.isVisible = isVisible;
+    }
+
+    public static String convertProjectTypesToString(List<ProjectType> proposalTypes) {
+        if (proposalTypes == null || proposalTypes.isEmpty()) {
+            return "";
+        }
+        return proposalTypes.stream()
+                .map(Enum::name)
+                .collect(Collectors.joining(","));
+    }
+
+    public static List<ProjectType> convertStringToProjectTypes(String proposalTypes) {
+        if (proposalTypes == null || proposalTypes.isEmpty()) {
+            return new ArrayList<>();
+        }
+        return Arrays.stream(proposalTypes.split(","))
+                .map(ProjectType::valueOf)
+                .collect(Collectors.toList());
+    }
+
+    public String getDisplayTitle(User requestUser) {
+        return shouldHideContent(requestUser) ? HIDDEN_TITLE : this.title;
+    }
+
+    public String getDisplayContent(User requestUser) {
+        return shouldHideContent(requestUser) ? HIDDEN_CONTENT : this.content;
+    }
+
+    public String getDisplayEmail(User requestUser) {
+        return isAnonymous && !isAuthorized(requestUser) ? HIDDEN_EMAIL : this.email;
+    }
+
+    public String getDisplayWebsite(User requestUser) {
+        return isAnonymous && !isAuthorized(requestUser) ? HIDDEN_WEBSITE : this.webSite;
+    }
+
+    public String getDisplayUser(User requestUser) {
+        return (isAnonymous || !isVisible) && !isAuthorized(requestUser) ? HIDDEN_USER : user.getName();
+    }
+
+    private boolean shouldHideContent(User requestUser) {
+        return !isVisible && !isAuthorized(requestUser);
+    }
+
+    public boolean isAuthorized(User requestUser) {
+        return requestUser.getUserType() == UserType.ADMIN || this.user.equals(requestUser);
+    }
```

Comment:
```suggestion
    public boolean isAuthorized(User requestUser) {
        return requestUser.getUserType() == UserType.ADMIN || this.user.getId().equals(requestUser.getId());
    }
```

조회 부분 테스트하는데 COMPANY **유저 본인이 작성한 과제 제시글**이 `비밀글`, `익명 사용자`로 표시되는 오류가 있었습니다.
아마 여기서 두 객체 equals가 의도대로 작동이 안되는 것 같아요! 위처럼 수정하니까 정상 작동 되었습니다...!!

Reply:
아 그런데 생각해보니까 만약 유저가 탈퇴하면 과제제안 작성자는 setnull로 수정되어서 제가 쓴 것처럼 하면 `this.user.getId()`에서 오류 터질 수도 있을 것 같아요...! null 체크를 앞에 추가하거나 다른 방식으로 해야될 것 같습니다

Reply:
[55dc505]